repository/src/test/java/org/alfresco/repo/security/authentication/identityservice/ClientRegistrationProviderUnitTest.java

```diff
@@ -0,0 +1,263 @@
+        when(responseEntity.hasBody()).thenReturn(true);
+        when(responseEntity.getBody()).thenReturn("");
+
+        oidcResponse = spy(OIDCProviderMetadata.parse(OPENID_CONFIGURATION));
+    }
+
+    @Test
+    public void shouldCreateClientRegistration()
+    {
+        config.setIssuerUrl("https://login.serviceonline.com/alfresco/v2.0");
+        try (MockedStatic<OIDCProviderMetadata> providerMetadata = Mockito.mockStatic(OIDCProviderMetadata.class))
+        {
+            providerMetadata.when(() -> OIDCProviderMetadata.parse(any(String.class))).thenReturn(oidcResponse);
+
+            ClientRegistration clientRegistration = new ClientRegistrationProvider(config).createClientRegistration(
+                restTemplate);
+            assertThat(clientRegistration).isNotNull();
+            assertThat(clientRegistration.getClientId()).isNotNull();
+            assertThat(clientRegistration.getProviderDetails().getAuthorizationUri()).isNotNull();
+            assertThat(clientRegistration.getProviderDetails().getTokenUri()).isNotNull();
+            assertThat(clientRegistration.getProviderDetails().getJwkSetUri()).isNotNull();
+            assertThat(clientRegistration.getProviderDetails().getUserInfoEndpoint()).isNotNull();
+            assertThat(clientRegistration.getProviderDetails().getIssuerUri()).isNotNull();
+            assertThat(requestEntityCaptor.getValue().getUrl().toASCIIString()).isEqualTo(
+                AUTH_SERVER + DISCOVERY_PATH_SEGEMENTS);
+        }
+    }
+
+    @Test
+    public void shouldCreateClientRegistrationWithoutIssuerConfigured()
+    {
+        config.setIssuerUrl(null);
+        try (MockedStatic<OIDCProviderMetadata> providerMetadata = Mockito.mockStatic(OIDCProviderMetadata.class))
+        {
+            providerMetadata.when(() -> OIDCProviderMetadata.parse(any(String.class))).thenReturn(oidcResponse);
+
+            ClientRegistration clientRegistration = new ClientRegistrationProvider(config).createClientRegistration(
+                restTemplate);
+            assertThat(clientRegistration).isNotNull();
+            assertThat(clientRegistration.getClientId()).isNotNull();
+            assertThat(clientRegistration.getProviderDetails().getAuthorizationUri()).isNotNull();
+            assertThat(clientRegistration.getProviderDetails().getTokenUri()).isNotNull();
+            assertThat(clientRegistration.getProviderDetails().getJwkSetUri()).isNotNull();
+            assertThat(clientRegistration.getProviderDetails().getUserInfoEndpoint()).isNotNull();
+            assertThat(clientRegistration.getProviderDetails().getIssuerUri()).isNotNull();
+            assertThat(requestEntityCaptor.getValue().getUrl().toASCIIString()).isEqualTo(
+                AUTH_SERVER + DISCOVERY_PATH_SEGEMENTS);
+        }
+    }
+
+    @Test
+    public void shouldThrowIdentityServiceExceptionIfIssuerIsNotValid()
+    {
+        config.setIssuerUrl("https://invalidissuer.com");
```

Comment:
How does this test work? I thought the underlying RestTemplate was mocked but I don't see any special mocking for this usecase.
If it's actually doing calls to this `https://invalidissuer.com`, can we replace it instead with something else that's safer? E.g.: some completely gibberish string that is not compatible with the URL syntax or e.g.: `https://hyland.com` if it's absolutely necessary?

Reply:
This line mocks the RestTemplate to return always the 200 OK status code with an empty body.  https://github.com/Alfresco/alfresco-community-repo/pull/2426/files#diff-f7fe0229c772ae958a59d19f8011a277d6ee5c62f511558aced4a975476514c0R76
This test also mocks the static method - OIDCProviderMetadata.parse to parse a prepared openid discovery document for tests.

Reply:
If it's all mocked anyways then let's please use an invalid syntax for the domain name such as `https://invalid-issuer` so that there is 0% chance of hitting a real domain in case something in the implementation changes and the mocks are not updated appropriately.

Reply:
Done.